### ISSUE TITLE
game crash when autoreplacing broken tools from the crafting output

### ISSUE THREAD

**Issue by wasakoki** [anegostudios/VintageStory-Issues#2741]
### Game Version

1.18.3

### Platform

Linux

### Modded

Vanilla

### SP/MP

Multiplayer

### Description

when a tool breaks, if you have your crafting menu open the game tries to replace your broken tool with the item in your crafting output grid, which crashes the game.

### How to reproduce

0. not needed, but it will probably help to turn on immersive mouse mode
1. get tool (my example will use flint axe)
2. put flint axe head and stick in crafting gui (recipe to make the previous step tool)
3. with this crafting menu open, break your already crafted axe
4. game crash

### Logs

```Running on 64 bit Linux with 15 GB. 
Game Version: v1.18.3 (Stable)
5/14/2023 12:34:00 PM: Critical error occurred
Loaded Mods: game@1.18.3, creative@1.18.3, survival@1.18.3
System.NullReferenceException: Object reference not set to an instance of an object
  at Vintagestory.Common.ItemSlotCraftingOutput.triggerEvent (Vintagestory.API.Common.ItemStack craftedStack, System.Int32 moved, Vintagestory.API.Common.IPlayer actingPlayer) [0x0001e] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Common.ItemSlotCraftingOutput.TryPutInto (Vintagestory.API.Common.ItemSlot sinkSlot, Vintagestory.API.Common.ItemStackMoveOperation& op) [0x0009b] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.API.Common.ItemSlot.TryPutInto (Vintagestory.API.Common.IWorldAccessor world, Vintagestory.API.Common.ItemSlot sinkSlot, System.Int32 quantity) [0x0000b] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.API.Common.CollectibleObject+<>c__DisplayClass78_0.<RefillSlotIfEmpty>b__0 (Vintagestory.API.Common.ItemSlot invslot) [0x00051] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.API.Common.EntityPlayer.WalkInventory (Vintagestory.API.Common.OnInventorySlot handler) [0x00067] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.API.Common.CollectibleObject.RefillSlotIfEmpty (Vintagestory.API.Common.ItemSlot slot, Vintagestory.API.Common.EntityAgent byEntity, Vintagestory.API.Common.ActionConsumable`1[T] matcher) [0x00029] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.API.Common.CollectibleObject.DamageItem (Vintagestory.API.Common.IWorldAccessor world, Vintagestory.API.Common.Entities.Entity byEntity, Vintagestory.API.Common.ItemSlot itemslot, System.Int32 amount) [0x0008d] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.API.Common.CollectibleObject.OnBlockBrokenWith (Vintagestory.API.Common.IWorldAccessor world, Vintagestory.API.Common.Entities.Entity byEntity, Vintagestory.API.Common.ItemSlot itemslot, Vintagestory.API.Common.BlockSelection blockSel, System.Single dropQuantityMultiplier) [0x000aa] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.GameContent.ItemAxe.OnBlockBrokenWith (Vintagestory.API.Common.IWorldAccessor world, Vintagestory.API.Common.Entities.Entity byEntity, Vintagestory.API.Common.ItemSlot itemslot, Vintagestory.API.Common.BlockSelection blockSel, System.Single dropQuantityMultiplier) [0x00072] in <9c8a60ff4e6a41b5bdf681805ea14b8e>:0 
  at Vintagestory.Client.NoObf.ClientMain.OnPlayerTryDestroyBlock (Vintagestory.API.Common.BlockSelection blockSelection) [0x0007e] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.ContinueBreakSurvival (Vintagestory.API.Common.BlockSelection blockSelection, Vintagestory.API.Common.Block block, System.Single dt) [0x001e3] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected (System.Single dt) [0x00488] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.UpdatePicking (System.Single dt) [0x0009e] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame (System.Single dt) [0x0004a] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.API.Client.DummyRenderer.OnRenderFrame (System.Single deltaTime, Vintagestory.API.Client.EnumRenderStage stage) [0x00000] in <ac8f793f72eb491b9577378f83f29660>:0 
  at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage (Vintagestory.API.Client.EnumRenderStage stage, System.Single dt) [0x00072] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage (Vintagestory.API.Client.EnumRenderStage stage, System.Single dt) [0x00028] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer (System.Single dt) [0x00093] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.GuiScreenRunningGame.RenderToDefaultFramebuffer (System.Single dt) [0x0003e] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.ScreenManager.Render (System.Single dt) [0x001ec] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.ScreenManager.OnNewFrame (System.Single dt) [0x000cc] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame (System.Object sender, OpenTK.FrameEventArgs e) [0x00141] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[OpenTK.FrameEventArgs].invoke_void_object_TEventArgs(object,OpenTK.FrameEventArgs)
  at OpenTK.GameWindow.OnRenderFrame (OpenTK.FrameEventArgs e) [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.OnRenderFrameInternal (OpenTK.FrameEventArgs e) [0x00010] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.RaiseRenderFrame (System.Double elapsed, System.Double& timestamp) [0x0000c] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.DispatchRenderFrame () [0x00038] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run (System.Double updates_per_second, System.Double frames_per_second) [0x00104] in <a4886e33f9fc416195d24950e7205b94>:0 
  at OpenTK.GameWindow.Run () [0x00000] in <a4886e33f9fc416195d24950e7205b94>:0 
  at Vintagestory.Client.ClientProgram.Start (Vintagestory.Client.ClientProgramArgs args, System.String[] rawArgs) [0x00421] in <195b02f48a2144898140669d265397a6>:0 
  at Vintagestory.Client.ClientProgram+<>c__DisplayClass16_0.<.ctor>b__0 () [0x00000] in <195b02f48a2144898140669d265397a6>:0 
  at Vintagestory.ClientNative.CrashReporter.Start (System.Threading.ThreadStart start) [0x00007] in <7dc1ecd263244f3984ddc1d2817eedfa>:0 
```

**Comment by Craluminum2413** [comment]
@DArkHekRoMaNT are you able to reproduce this?

**Comment by DArkHekRoMaNT** [comment]
> @DArkHekRoMaNT are you able to reproduce this?

Yes it crashes

**Comment by Th3Dilli** [comment]
thanks for reporting

should be fixed with the next release (most likely 1.19)
